hasher-matcher-actioner/hmalib/lambdas/hashing.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import functools
+from hmalib.common.content_sources import URLContentSource
+import boto3
+import os
+import json
+
+from mypy_boto3_dynamodb import DynamoDBServiceResource
+from mypy_boto3_sqs import SQSClient
+
+from threatexchange.content_type.meta import get_content_type_for_name
+from threatexchange.content_type.photo import PhotoContent
+from threatexchange.content_type.video import VideoContent
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.pdq import PdqSignal
+
+from hmalib.common.logging import get_logger
+from hmalib import metrics
+from hmalib.common.message_models import URLSubmissionMessage
+from hmalib.hashing import Hasher
+
+logger = get_logger(__name__)
+sqs_client = boto3.client("sqs")
+
+
+@functools.lru_cache(maxsize=None)
+def get_dynamodb() -> DynamoDBServiceResource:
+    return boto3.resource("dynamodb")
+
+
+@functools.lru_cache(maxsize=None)
+def get_sqs_client() -> SQSClient:
+    return boto3.client("sqs")
+
+
+# OUTPUT_QUEUE_URL = os.environ["HASHES_QUEUE_URL"]
+DYNAMODB_TABLE = os.environ["DYNAMODB_TABLE"]
+
+# If you want to support additional content or signal types, they can be added
+# here.
+hasher = Hasher(
+    supported_content_types=[PhotoContent, VideoContent],
+    supported_signal_types=[PdqSignal, VideoMD5Signal],
+)
+
+
+def lambda_handler(event, context):
+    """
+    SQS Events generated by the submissions API or by files being added to S3.
+    Downloads files to temp-storage, identifies content_type and generates
+    allowed signal_types from it.
+
+    Saves hash output to DynamoDB, sends a message on an output queue.
+
+    Note that this brings the contents of a file into memory. This is subject to
+    the resource limitation on the lambda. Potentially extendable until 10GB, but
+    that would be super-expensive. [1]
+
+    [1]: https://docs.aws.amazon.com/lambda/latest/dg/configuration-console.html
+    """
+    records_table = get_dynamodb().Table(DYNAMODB_TABLE)
+
+    for sqs_record in event["Records"]:
+        logger.info(f"Body: {sqs_record}")
+        message = json.loads(sqs_record["body"])
+
+        if message.get("Event") == "s3:TestEvent":
+            continue
+
+        if URLSubmissionMessage.could_be(message):
```

Comment:
Why not feed S3 events here as well? Why do they still go to the pdq_hasher?

Reply:
Reducing the surface area of change. Once proven out, we can move all submission types here.